torch/nested/_internal/nested_tensor.py

```diff
@@ -3,23 +3,224 @@
 import torch
 from torch._C import DispatchKey, DispatchKeySet
 from torch._prims_common import is_expandable_to
-from torch.fx.experimental.symbolic_shapes import has_free_symbols
 from torch.utils.weak import WeakTensorKeyDictionary
 from typing import *  # noqa: F403
+import weakref
 
-_tensor_id_counter = 0
-_tensor_symint_registry = WeakTensorKeyDictionary()
+def _get_nested_int(equiv_set, vec):
+    return torch._C._get_nested_int(equiv_set, coeff=1, vec=vec)
 
+class NestedIntRegistry():
+    # Class to manage the association between 1d tensors (vec) and nested ints.
+    # Use this class to (1) obtain nested int from vec (2) read/write metadata
+    # associated with vecs/nested ints
+    #
+    # Pass a vec to maybe_create to create a new nested int. Multiple nested int
+    # can be associated with a vec. vec are then grouped into equivalence sets
+    # so that comparisons between the vec's nested ints can be made by comparing
+    # their equiv set. Morally two vec are in the same equiv set if and only if
+    # they have the same data, e.g., vec.cpu() and vec.cuda() belong to the same
+    # equiv set even though they are on different devices because they have the
+    # same data.
```

Comment:
Is this moral behavior (same data -> same equiv set) enforced or is it the case that vecs are only part of the same equiv set if this is done explicitly? I think it's good to be clear on what's enforced or not here

Reply:
Not enforced, will update doc to make that clearer.